build.gradle

```diff
@@ -68,7 +68,7 @@ sourceCompatibility = 1.8
 targetCompatibility = 1.8
 
 tasks.withType(JavaCompile) {
-    options.compilerArgs += [ '-Xlint:all' ]
+    options.compilerArgs += [ '-Xlint:all', '-XepExcludedPaths:.*\\/src\\/(integ_)?test\\/.*', '-Xep:ReferenceEquality:OFF' ]
```

Comment:
I'd recommend not turning off ReferenceEquality for now.  I agree there may be some false negatives related to enums (which should probably be reported back to Error Prone), but I think we should take our time evaluating them rather than simply disabling it and possibly forgetting that we did so.  If we agree it is not providing much value, then we can disable it en masse.

Disabling Error Prone for all tests is probably overkill.  We can just turn off ClassCanBeStatic for unit test code with the following change:

```diff
diff --git a/build.gradle b/build.gradle
index 112a3bfcf..a053fe14f 100644
--- a/build.gradle
+++ b/build.gradle
@@ -68,7 +68,7 @@ sourceCompatibility = 1.8
 targetCompatibility = 1.8
 
 tasks.withType(JavaCompile) {
-    options.compilerArgs += [ '-Xlint:all', '-XepExcludedPaths:.*\\/src\\/(integ_)?test\\/.*', '-Xep:ReferenceEquality:OFF' ]
+    options.compilerArgs += [ '-Xlint:all' ]
     options.incremental = true
     options.encoding = 'UTF-8'
 }
@@ -303,6 +303,10 @@ checkstyleTest {
     source sourceSets.test.output.resourcesDir
 }
 
+compileTestJava {
+    options.compilerArgs += [ '-Xep:ClassCanBeStatic:OFF' ]
+}
+
 afterEvaluate {
     def junitPlatformTest = tasks.junitPlatformTest
 
```

It would be great if Error Prone supported disabling individual rules by path (so we could apply the filter to only `*Test` classes), but that doesn't appear possible in the current version.